### PR TITLE
CLDR-13857 add XMLSource cache

### PR DESCRIPTION
**Pull request by AlintaLu**
- Add XMLSource cache
- Add CLDRFile new constructor

##### Checklist

- [ ] Issue filed: https://unicode-org.atlassian.net/browse/CLDR-13857
- [ ] Updated PR title and link in previous line to include Issue number

### CODE REVIEW

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 1]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+
+        private String orderingAttribute() {
+            return "[@_q=\"" + (orderedCounter[level]++) + "\"]";
+        }
+
+        private void putAndFixDeprecatedAttribute(String element, String attribute, String value) {
+            if (attribute.equals("draft")) {
+                if (value.equals("true"))
+                    value = "approved";
+                else if (value.equals("false")) value = "unconfirmed";
```
Comment (mihnita) [inline tools/java/org/unicode/cldr/util/XMLSource.java:2004]: nitpick: line break before `value = "unconfirmed"`

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 2]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+            this.minimalDraftStatus = minimalDraftStatus;
+        }
+
+        private String show(Attributes attributes) {
+            if (attributes == null) return "null";
+            String result = "";
+            for (int i = 0; i < attributes.getLength(); ++i) {
+                String attribute = attributes.getQName(i);
+                String value = attributes.getValue(i);
+                result += "[@" + attribute + "=\"" + value + "\"]"; // TODO quote the value??
```
Comment (mihnita) [inline tools/java/org/unicode/cldr/util/XMLSource.java:1933]: Repeated String concatenation is slow, use StringBuilder.

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 3]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+            source.freeze();
+            return source;
+        }
+
+        List<File> dirList = new ArrayList<>();
+        for (File dir: key.dirs) {
+            dirList.clear();
+            dirList.add(dir);
+            XMLSourceCacheKey singleKey = new XMLSourceCacheKey(key.localeId, dirList, key.minimalDraftStatus);
+            XMLSource singleSource = null;
```
Comment (mihnita) [inline tools/java/org/unicode/cldr/util/XMLSource.java:1850]: ``
XMLSource singleSource = cache.getUnchecked(singleKey);
```

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 4]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+            singleSource = cache.getUnchecked(singleKey);
+            list.add(singleSource);
```
Comment (mihnita) [inline tools/java/org/unicode/cldr/util/XMLSource.java:1852]: If `singleKey` is no in `cache` then we add `null` to `list`?

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 5]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+                    String value = attributes.getValue(i);
+
+                    if (attribute.equals("cldrVersion")
+                        && (qName.equals("version"))) {
+                        ((SimpleXMLSource) source).setDtdVersionInfo(VersionInfo.getInstance(value));
+                    } else {
+                        putAndFixDeprecatedAttribute(qName, attribute, value);
+                    }
+                }
+                for (Iterator<String> it = attributeOrder.keySet().iterator(); it.hasNext();) {
```
Comment (mihnita) [inline tools/java/org/unicode/cldr/util/XMLSource.java:1971]: Nitpick: a more common pattern is
```
for (Entry<String, String> entry : attributeOrder.entrySet()) {
    // use entry.getKey() and entry.getValue()
}
```

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 6]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+                    String attribute = it.next();
+                    String value = attributeOrder.get(attribute);
+                    String both = "[@" + attribute + "=\"" + value + "\"]"; // TODO quote the value??
+                    currentFullXPath += both;
```
Comment (mihnita) [inline tools/java/org/unicode/cldr/util/XMLSource.java:1975]: If there are many string concatenations in a loop consider StringBuilder

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 7]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+                }
+            }
+
+            attributeOrder.put(attribute, value);
+        }
+
+        /**
+         * Types which changed from 'type' to 'choice', but not in supplemental data.
+         */
+        private static Set<String> changedTypes = new HashSet<>(Arrays.asList(new String[] {
```
Comment (mihnita) [inline tools/java/org/unicode/cldr/util/XMLSource.java:2018]: Make it `final` and uppercase (`CHANGED_TYPES = ...`)
And move it at the top of the class, all 3 of them..

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 8]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+                    source.addComment(lastLeafNode, comment, XPathParts.Comments.CommentType.POSTBLOCK);
+                    comment = null;
+                }
+            }
+            // currentXPath = stripAfter(currentXPath, qName);
+            currentFullXPath = stripAfter(currentFullXPath, qName);
+            justPopped = true;
+        }
+
+        static Pattern WHITESPACE_WITH_LF = PatternCache.get("\\s*\\u000a\\s*");
```
Comment (mihnita) [inline tools/java/org/unicode/cldr/util/XMLSource.java:2099]: `static final` ?
And move all 3 of these at the top of the class.

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 9]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+
+        /**
+         * Trim leading whitespace if there is a linefeed among them, then the same with trailing.
+         *
+         * @param source
+         * @return
+         */
+        private String trimWhitespaceSpecial(String source) {
+            if (DEBUG && CONTROLS.containsSome(source)) {
+                System.out.println("*** " + source);
```
Comment (mihnita) [inline tools/java/org/unicode/cldr/util/XMLSource.java:2111]: Can we log instead?

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 10]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+            if (!source.contains("\n")) {
+                return source;
+            }
+            source = whitespaceWithLf.reset(source).replaceAll("\n");
+            return source;
+        }
+
+        private void warnOnOverride(String former, String formerPath) {
+            String distinguishing = CLDRFile.getDistinguishingXPath(formerPath, null);
+            System.out.println("\tERROR in " + source.getLocaleID()
```
Comment (mihnita) [inline tools/java/org/unicode/cldr/util/XMLSource.java:2122]: Log instead?

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 11]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+                + ";\toverriding old value <" + former + "> at path " + distinguishing +
+                "\twith\t<" + lastChars + ">" +
+                CldrUtility.LINE_SEPARATOR + "\told fullpath: " + formerPath +
+                CldrUtility.LINE_SEPARATOR + "\tnew fullpath: " + currentFullXPath);
+            overrideCount += 1;
```
Comment (mihnita) [inline tools/java/org/unicode/cldr/util/XMLSource.java:2127]: `overrideCount++`

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 12]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+            }
+            value = trimWhitespaceSpecial(value);
+            source.add(fullXPath, value);
+        }
+
+        private void pop(String qName) {
+            Log.logln(LOG_PROGRESS, "pop\t" + qName);
+            --level;
+
+            if (lastChars.length() != 0 || justPopped == false) {
```
Comment (mihnita) [inline tools/java/org/unicode/cldr/util/XMLSource.java:2049]: `!lastChars.isEmpty()`

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 13]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+        public void characters(char[] ch, int start, int length)
+            throws SAXException {
+            try {
+                String value = new String(ch, start, length);
+                Log.logln(LOG_PROGRESS, "characters:\t" + value);
+                // we will strip leading and trailing line separators in another place.
+                // if (value.indexOf(XML_LINESEPARATOR) >= 0) {
+                // value = value.replace(XML_LINESEPARATOR, '\u0020');
+                // }
+                lastChars += value;
```
Comment (mihnita) [inline tools/java/org/unicode/cldr/util/XMLSource.java:2227]: if `characters` is called a lot then consider making `lastChars` a `StringBuilder`

In general I would search for all instances of `+=`

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 14]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+            Log.logln(LOG_PROGRESS, "endDocument");
+            try {
+                if (comment != null) source.addComment(null, comment, XPathParts.Comments.CommentType.LINE);
+            } catch (RuntimeException e) {
+                e.printStackTrace();
+                throw e;
+            }
+        }
+
+        // ==== The following are just for debuggin =====
```
Comment (mihnita) [inline tools/java/org/unicode/cldr/util/XMLSource.java:2301]: Nitpick: `debugging` :-)

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 15]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+            this.getXpathComments().addComment(type, xpath, comment);
+        }
+        return this;
+    }
+
+    /**
+     * Get a string value from an xpath.
+     */
+    public String getStringValue(String xpath) {
+            String result = this.getValueAtPath(xpath);
```
Comment (mihnita) [inline tools/java/org/unicode/cldr/util/XMLSource.java:2433]: nitpick: `return this.getValueAtPath(xpath);`

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 16]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+    public DtdType getSimpleXMLSourceDtdType() {
+        return dtdType;
+    }
+
+    public void setDtdType(DtdType dtdType) {
+        this.dtdType = dtdType;
+    }
+
+    private static class XMLSourceCacheKey {
+        String localeId;
```
[the remark this answers is not in the document]
Comment (AlintaLu) [reply tools/java/org/unicode/cldr/util/XMLSource.java:1778]: Make XMLSourceCacheKey final

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 17]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+                return false;
+            }
+            XMLSourceCacheKey other = (XMLSourceCacheKey) obj;
+            if (!Objects.equals(dirs, other.dirs)) {
+                return false;
+            }
+            if (minimalDraftStatus != other.minimalDraftStatus) {
+                return false;
+            }
+            if (localeId == null) {
```
[the remark this answers is not in the document]
Comment (macchiati) [reply tools/java/org/unicode/cldr/util/XMLSource.java:1812]: Objects.equals does the following:

    public static boolean equals(Object a, Object b) {
        return (a == b) || (a != null && a.equals(b));
    }

We have to check for null before we do other.localeId (Yiqing saw this). Might be able to clean up, but should do later.

File: tools/java/org/unicode/cldr/util/XMLSource.java (hunk 3 above)
Comment (AlintaLu) [reply tools/java/org/unicode/cldr/util/XMLSource.java:1850]: Agree and will change.

File: tools/java/org/unicode/cldr/util/XMLSource.java (hunk 4 above)
Comment (AlintaLu) [reply tools/java/org/unicode/cldr/util/XMLSource.java:1852]: will check null before put into cache

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 18]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+        source = list.get(0).cloneAsThawed();
+        for (int i = 1; i < list.size(); i++) {
+            XMLSource other = list.get(i);
+            source.putAll(other, 0); // 0 --> merge_keep_mine
+            source.getXpathComments().joinAll(other.getXpathComments());
+        }
+        source.freeze();
+        return source;
+    }
+    public static XMLSource getFrozenInstance(String localeId, List<File> dirs, DraftStatus minimalDraftStatus) {
```
[the remark this answers is not in the document]
Comment (AlintaLu) [reply tools/java/org/unicode/cldr/util/XMLSource.java:1864]: Agree and will change

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 19]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+        for (File dir: key.dirs) {
+            dirList.clear();
+            dirList.add(dir);
+            XMLSourceCacheKey singleKey = new XMLSourceCacheKey(key.localeId, dirList, key.minimalDraftStatus);
+            XMLSource singleSource = null;
+            singleSource = cache.getUnchecked(singleKey);
+            list.add(singleSource);
+        }
+
+        source = list.get(0).cloneAsThawed();
```
[the remark this answers is not in the document]
Comment (macchiati) [reply tools/java/org/unicode/cldr/util/XMLSource.java:1855]: True, however the caller should never do that, and getting that exception will alert the caller properly. Slightly better to check for empty dirs and change to different message, but not worth doing now.

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 20]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+        }
+        source.freeze();
+        return source;
+    }
+    public static XMLSource getFrozenInstance(String localeId, List<File> dirs, DraftStatus minimalDraftStatus) {
+        XMLSourceCacheKey key = new XMLSourceCacheKey(localeId, dirs, minimalDraftStatus);
+        return cache.getUnchecked(key);
+    }
+
+    public static XMLSource loadXMLFile (File f, String localeId, DraftStatus minimalDraftStatus) {
```
[the remark this answers is not in the document]
Comment (AlintaLu) [reply tools/java/org/unicode/cldr/util/XMLSource.java:1869]: Agree and will change

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 21]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+                + XML_HANDLER.overrideCount
+                + "; The exact problems are printed on the console above.");
+        }
+        return source;
+    }
+
+    private static boolean LOG_PROGRESS = false;
+    private static final boolean DEBUG = false;
+
+    private static class XMLNormalizingHandler implements AllHandler {
```
[the remark this answers is not in the document]
Comment (macchiati) [reply tools/java/org/unicode/cldr/util/XMLSource.java:1902]: This is code that was moved from another place, so some of these are problems that already existed. I think this is adequately tested by the current suite of tests for CLDRFile.

[the remark this answers is not in the document]
Comment (macchiati) [reply tools/java/org/unicode/cldr/util/XMLSource.java:1902]: That being said, we should file a ticket to add a unit test for this.

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 22]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+        private boolean justPopped = false;
+        private String lastChars = "";
+        private String currentFullXPath = "/";
+        private String comment = null;
+        private Map<String, String> attributeOrder;
+        private DtdData dtdData;
+        private XMLSource source;
+        private String lastActiveLeafNode;
+        private String lastLeafNode;
+        private int isSupplemental = -1;
```
[the remark this answers is not in the document]
Comment (AlintaLu) [reply tools/java/org/unicode/cldr/util/XMLSource.java:1916]: Will change to an Enum

[the remark this answers is not in the document]
Comment (AlintaLu) [reply tools/java/org/unicode/cldr/util/XMLSource.java:1916]: like NEVER_SET, IS_SUMPPLEMENTAL, NOT_SUPPLEMENTAL

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 23]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+        private int[] orderedCounter = new int[30]; // just make deep enough to handle any CLDR file.
```
[the remark this answers is not in the document]
Comment (macchiati) [reply tools/java/org/unicode/cldr/util/XMLSource.java:1917]: If it overflows, we will find out!

File: tools/java/org/unicode/cldr/util/XMLSource.java (hunk 2 above)
Comment (AlintaLu) [reply tools/java/org/unicode/cldr/util/XMLSource.java:1933]: Agree and will change

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 24]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+            return result;
+        }
+
+        private void push(String qName, Attributes attributes) {
+            // SHOW_ALL &&
+            Log.logln(LOG_PROGRESS, "push\t" + qName + "\t" + show(attributes));
+            ++level;
+            if (!qName.equals(orderedString[level])) {
+                // orderedCounter[level] = 0;
+                orderedString[level] = qName;
```
[the remark this answers is not in the document]
Comment (macchiati) [reply tools/java/org/unicode/cldr/util/XMLSource.java:1944]: ok to just throw

File: tools/java/org/unicode/cldr/util/XMLSource.java (hunk 5 above)
Comment (macchiati) [reply tools/java/org/unicode/cldr/util/XMLSource.java:1971]: Yes, that is very old code, predated "  :  ". Should be fixed.

File: tools/java/org/unicode/cldr/util/XMLSource.java (hunk 6 above)
Comment (macchiati) [reply tools/java/org/unicode/cldr/util/XMLSource.java:1975]: Agreed.

Comment (AlintaLu) [reply tools/java/org/unicode/cldr/util/XMLSource.java:1975]: Agree and will change.

File: tools/java/org/unicode/cldr/util/XMLSource.java (hunk 1 above)
Comment (AlintaLu) [reply tools/java/org/unicode/cldr/util/XMLSource.java:2004]: Will change

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 25]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+                }
+                comment = null;
+            }
+            justPopped = false;
+            lastActiveLeafNode = null;
+            Log.logln(LOG_PROGRESS, "currentFullXPath\t" + currentFullXPath);
+        }
+
+        private String orderingAttribute() {
+            return "[@_q=\"" + (orderedCounter[level]++) + "\"]";
```
[the remark this answers is not in the document]
Comment (macchiati) [reply tools/java/org/unicode/cldr/util/XMLSource.java:1997]: orderedCounter, orderedString, and level logically form a single class that allows adding elements, but never removed. So we don't care what the original contents were, as long as nothing is accessed beyond level.

Suggest just documenting that at this point.

File: tools/java/org/unicode/cldr/util/XMLSource.java (hunk 1 above)
Comment (macchiati) [reply tools/java/org/unicode/cldr/util/XMLSource.java:2004]: Yes, insert {
...
}

File: tools/java/org/unicode/cldr/util/XMLSource.java (hunk 7 above)
Comment (AlintaLu) [reply tools/java/org/unicode/cldr/util/XMLSource.java:2018]: Agree and will change.

File: tools/java/org/unicode/cldr/util/XMLSource.java (hunk 12 above)
Comment (AlintaLu) [reply tools/java/org/unicode/cldr/util/XMLSource.java:2049]: Agree and will change

Comment (macchiati) [reply tools/java/org/unicode/cldr/util/XMLSource.java:2049]: Was written before java had isEmpty!

File: tools/java/org/unicode/cldr/util/XMLSource.java (hunk 8 above)
Comment (macchiati) [reply tools/java/org/unicode/cldr/util/XMLSource.java:2099]: Pattern can be moved, and the UnicodeSet moved (AND)

new UnicodeSet("[:cc:]").freeze();

Matcher shouldn't be static. Is mutable.

Comment (AlintaLu) [reply tools/java/org/unicode/cldr/util/XMLSource.java:2099]: Will change

File: tools/java/org/unicode/cldr/util/XMLSource.java (hunk 11 above)
Comment (macchiati) [reply tools/java/org/unicode/cldr/util/XMLSource.java:2127]: compiler will do the same with these, so just a matter of style.

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 26]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+            return input.substring(0, pos);
+        }
+
+        private static int findLastSlash(String input) {
+            int braceStack = 0;
+            char inQuote = 0;
+            for (int i = input.length() - 1; i >= 0; --i) {
+                char ch = input.charAt(i);
+                switch (ch) {
+                case '\'':
```
[the remark this answers is not in the document]
Comment (AlintaLu) [reply tools/java/org/unicode/cldr/util/XMLSource.java:2147]: Will do it.

File: tools/java/org/unicode/cldr/util/XMLSource.java (hunk 13 above)
Comment (AlintaLu) [reply tools/java/org/unicode/cldr/util/XMLSource.java:2227]: Agree and will change

Comment (macchiati) [reply tools/java/org/unicode/cldr/util/XMLSource.java:2227]: Can't do; this is in the XML parsing code (external).

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 27]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+                throw e;
+            }
+        }
+
+        @Override
+        public void startDTD(String name, String publicId, String systemId) throws SAXException {
+            Log.logln(LOG_PROGRESS, "startDTD name: " + name
+                + ", publicId: " + publicId
+                + ", systemId: " + systemId);
+            commentStack++;
```
[the remark this answers is not in the document]
Comment (AlintaLu) [reply tools/java/org/unicode/cldr/util/XMLSource.java:2240]: Agree, will change to commentStackIndex

[the remark this answers is not in the document]
Comment (AlintaLu) [reply tools/java/org/unicode/cldr/util/XMLSource.java:2240]: Will change to commentStackIndex

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 28]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+        public void comment(char[] ch, int start, int length) throws SAXException {
+            final String string = new String(ch, start, length);
+            Log.logln(LOG_PROGRESS, commentStack + " comment " + string);
+            try {
+                if (commentStack != 0) return;
+                String comment0 = trimWhitespaceSpecial(string).trim();
+                if (lastActiveLeafNode != null) {
+                    source.addComment(lastActiveLeafNode, comment0, XPathParts.Comments.CommentType.LINE);
+                } else {
+                    comment = (comment == null ? comment0 : comment + XPathParts.NEWLINE + comment0);
```
[the remark this answers is not in the document]
Comment (AlintaLu) [reply tools/java/org/unicode/cldr/util/XMLSource.java:2261]: Will file a ticket to see if there is a performance improvement if use StringJoiner/StringBuilder.

File: tools/java/org/unicode/cldr/util/XMLSource.java (hunk 14 above)
Comment (AlintaLu) [reply tools/java/org/unicode/cldr/util/XMLSource.java:2301]: Agree

File: tools/java/org/unicode/cldr/util/XMLSource.java (hunk 15 above)
Comment (AlintaLu) [reply tools/java/org/unicode/cldr/util/XMLSource.java:2433]: Agree, will refractor

File: tools/java/org/unicode/cldr/util/XMLSource.java [hunk 29]
```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
...
+        }
+        return this;
+    }
+
+    /**
+     * Show a SAX exception in a readable form.
+     */
+    public static String showSAX(SAXParseException exception) {
+        return exception.getMessage()
+            + ";\t SystemID: " + exception.getSystemId()
```
[the remark this answers is not in the document]
Comment (AlintaLu) [reply tools/java/org/unicode/cldr/util/XMLSource.java:2468]: May not change it, it stays in purpose

File: tools/java/org/unicode/cldr/util/XMLSource.java (hunk 16 above)
[the remark this answers is not in the document]
Comment (AlintaLu) [reply tools/java/org/unicode/cldr/util/XMLSource.java:1778]: Agree

**Comment by AlintaLu** [comment]
Close it and please see https://github.com/unicode-org/cldr/pull/607 instead.

**Pull request closed without merging**